(jacocoTestReport) Build failure with JDK 21

### Current Behavior

    FAILURE: Build failed with an exception.

    * What went wrong:
    Execution failed for task ':jacocoTestReport'.
    > A failure occurred while executing org.gradle.internal.jacoco.JacocoReportAction
       > Could not create service of type IsolatedAntBuilder using BuildScopeServices.createIsolatedAntBuilder().
          > Could not inspect the Groovy system for ClassLoader VisitableURLClassLoader(tooling-implementation-loader)

    * Try:
    > Run with --debug option to get more log output.
    > Run with --scan to generate a Build Scan (Powered by Develocity).
    > Get more help at https://help.gradle.org.

    * Exception is:
    org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':jacocoTestReport'.
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.lambda$executeIfValid$1(ExecuteActionsTaskExecuter.java:135)
        at org.gradle.internal.Try$Failure.ifSuccessfulOrElse(Try.java:288)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeIfValid(ExecuteActionsTaskExecuter.java:133)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:121)
        at org.gradle.api.internal.tasks.execution.ProblemsTaskPathTrackingTaskExecuter.execute(ProblemsTaskPathTrackingTaskExecuter.java:41)
        at org.gradle.api.internal.tasks.execution.FinalizePropertiesTaskExecuter.execute(FinalizePropertiesTaskExecuter.java:46)
        at org.gradle.api.internal.tasks.execution.ResolveTaskExecutionModeExecuter.execute(ResolveTaskExecutionModeExecuter.java:51)
        at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:57)
        at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:74)
        at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:36)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.executeTask(EventFiringTaskExecuter.java:77)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:55)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:52)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:209)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:204)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:166)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:53)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:52)
        at org.gradle.execution.plan.LocalTaskNodeExecutor.execute(LocalTaskNodeExecutor.java:45)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:347)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:334)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.lambda$execute$0(DefaultTaskExecutionGraph.java:330)
        at org.gradle.internal.operations.CurrentBuildOperationRef.with(CurrentBuildOperationRef.java:84)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:330)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:319)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.execute(DefaultPlanExecutor.java:459)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.run(DefaultPlanExecutor.java:376)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
        at org.gradle.internal.concurrent.AbstractManagedExecutor$1.run(AbstractManagedExecutor.java:47)
    Caused by: org.gradle.workers.internal.DefaultWorkerExecutor$WorkExecutionException: A failure occurred while executing org.gradle.internal.jacoco.JacocoReportAction
        at org.gradle.workers.internal.DefaultWorkerExecutor$WorkItemExecution.waitForCompletion(DefaultWorkerExecutor.java:289)
        at org.gradle.internal.work.DefaultAsyncWorkTracker.lambda$waitForItemsAndGatherFailures$2(DefaultAsyncWorkTracker.java:130)
        at org.gradle.internal.Factories$1.create(Factories.java:31)
        at org.gradle.internal.work.DefaultWorkerLeaseService.withoutLocks(DefaultWorkerLeaseService.java:335)
        at org.gradle.internal.work.DefaultWorkerLeaseService.withoutLocks(DefaultWorkerLeaseService.java:318)
        at org.gradle.internal.work.DefaultWorkerLeaseService.withoutLock(DefaultWorkerLeaseService.java:323)
        at org.gradle.internal.work.DefaultAsyncWorkTracker.waitForItemsAndGatherFailures(DefaultAsyncWorkTracker.java:126)
        at org.gradle.internal.work.DefaultAsyncWorkTracker.waitForItemsAndGatherFailures(DefaultAsyncWorkTracker.java:92)
        at org.gradle.internal.work.DefaultAsyncWorkTracker.waitForAll(DefaultAsyncWorkTracker.java:78)
        at org.gradle.internal.work.DefaultAsyncWorkTracker.waitForCompletion(DefaultAsyncWorkTracker.java:66)
        at org.gradle.api.internal.tasks.execution.TaskExecution$3.run(TaskExecution.java:260)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:29)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:26)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:166)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.run(DefaultBuildOperationRunner.java:47)
        at org.gradle.api.internal.tasks.execution.TaskExecution.executeAction(TaskExecution.java:237)
        at org.gradle.api.internal.tasks.execution.TaskExecution.executeActions(TaskExecution.java:220)
        at org.gradle.api.internal.tasks.execution.TaskExecution.executeWithPreviousOutputFiles(TaskExecution.java:203)
        at org.gradle.api.internal.tasks.execution.TaskExecution.execute(TaskExecution.java:170)
        at org.gradle.internal.execution.steps.ExecuteStep.executeInternal(ExecuteStep.java:105)
        at org.gradle.internal.execution.steps.ExecuteStep.access$000(ExecuteStep.java:44)
        at org.gradle.internal.execution.steps.ExecuteStep$1.call(ExecuteStep.java:59)
        at org.gradle.internal.execution.steps.ExecuteStep$1.call(ExecuteStep.java:56)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:209)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:204)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:166)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:53)
        at org.gradle.internal.execution.steps.ExecuteStep.execute(ExecuteStep.java:56)
        at org.gradle.internal.execution.steps.ExecuteStep.execute(ExecuteStep.java:44)
        at org.gradle.internal.execution.steps.CancelExecutionStep.execute(CancelExecutionStep.java:42)
        at org.gradle.internal.execution.steps.TimeoutStep.executeWithoutTimeout(TimeoutStep.java:75)
        at org.gradle.internal.execution.steps.TimeoutStep.execute(TimeoutStep.java:55)
        at org.gradle.internal.execution.steps.PreCreateOutputParentsStep.execute(PreCreateOutputParentsStep.java:50)
        at org.gradle.internal.execution.steps.PreCreateOutputParentsStep.execute(PreCreateOutputParentsStep.java:28)
        at org.gradle.internal.execution.steps.RemovePreviousOutputsStep.execute(RemovePreviousOutputsStep.java:68)
        at org.gradle.internal.execution.steps.RemovePreviousOutputsStep.execute(RemovePreviousOutputsStep.java:38)
        at org.gradle.internal.execution.steps.BroadcastChangingOutputsStep.execute(BroadcastChangingOutputsStep.java:61)
        at org.gradle.internal.execution.steps.BroadcastChangingOutputsStep.execute(BroadcastChangingOutputsStep.java:26)
        at org.gradle.internal.execution.steps.CaptureOutputsAfterExecutionStep.execute(CaptureOutputsAfterExecutionStep.java:69)
        at org.gradle.internal.execution.steps.CaptureOutputsAfterExecutionStep.execute(CaptureOutputsAfterExecutionStep.java:46)
        at org.gradle.internal.execution.steps.ResolveInputChangesStep.execute(ResolveInputChangesStep.java:39)
        at org.gradle.internal.execution.steps.ResolveInputChangesStep.execute(ResolveInputChangesStep.java:28)
        at org.gradle.internal.execution.steps.BuildCacheStep.executeWithoutCache(BuildCacheStep.java:189)
        at org.gradle.internal.execution.steps.BuildCacheStep.lambda$execute$1(BuildCacheStep.java:75)
        at org.gradle.internal.Either$Right.fold(Either.java:176)
        at org.gradle.internal.execution.caching.CachingState.fold(CachingState.java:62)
        at org.gradle.internal.execution.steps.BuildCacheStep.execute(BuildCacheStep.java:73)
        at org.gradle.internal.execution.steps.BuildCacheStep.execute(BuildCacheStep.java:48)
        at org.gradle.internal.execution.steps.StoreExecutionStateStep.execute(StoreExecutionStateStep.java:46)
        at org.gradle.internal.execution.steps.StoreExecutionStateStep.execute(StoreExecutionStateStep.java:35)
        at org.gradle.internal.execution.steps.SkipUpToDateStep.executeBecause(SkipUpToDateStep.java:75)
        at org.gradle.internal.execution.steps.SkipUpToDateStep.lambda$execute$2(SkipUpToDateStep.java:53)
        at org.gradle.internal.execution.steps.SkipUpToDateStep.execute(SkipUpToDateStep.java:53)
        at org.gradle.internal.execution.steps.SkipUpToDateStep.execute(SkipUpToDateStep.java:35)
        at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsFinishedStep.execute(MarkSnapshottingInputsFinishedStep.java:37)
        at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsFinishedStep.execute(MarkSnapshottingInputsFinishedStep.java:27)
        at org.gradle.internal.execution.steps.ResolveIncrementalCachingStateStep.executeDelegate(ResolveIncrementalCachingStateStep.java:49)
        at org.gradle.internal.execution.steps.ResolveIncrementalCachingStateStep.executeDelegate(ResolveIncrementalCachingStateStep.java:27)
        at org.gradle.internal.execution.steps.AbstractResolveCachingStateStep.execute(AbstractResolveCachingStateStep.java:71)
        at org.gradle.internal.execution.steps.AbstractResolveCachingStateStep.execute(AbstractResolveCachingStateStep.java:39)
        at org.gradle.internal.execution.steps.ResolveChangesStep.execute(ResolveChangesStep.java:64)
        at org.gradle.internal.execution.steps.ResolveChangesStep.execute(ResolveChangesStep.java:35)
        at org.gradle.internal.execution.steps.ValidateStep.execute(ValidateStep.java:62)
        at org.gradle.internal.execution.steps.ValidateStep.execute(ValidateStep.java:40)
        at org.gradle.internal.execution.steps.AbstractCaptureStateBeforeExecutionStep.execute(AbstractCaptureStateBeforeExecutionStep.java:76)
        at org.gradle.internal.execution.steps.AbstractCaptureStateBeforeExecutionStep.execute(AbstractCaptureStateBeforeExecutionStep.java:45)
        at org.gradle.internal.execution.steps.AbstractSkipEmptyWorkStep.executeWithNonEmptySources(AbstractSkipEmptyWorkStep.java:136)
        at org.gradle.internal.execution.steps.AbstractSkipEmptyWorkStep.execute(AbstractSkipEmptyWorkStep.java:61)
        at org.gradle.internal.execution.steps.AbstractSkipEmptyWorkStep.execute(AbstractSkipEmptyWorkStep.java:38)
        at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsStartedStep.execute(MarkSnapshottingInputsStartedStep.java:38)
        at org.gradle.internal.execution.steps.LoadPreviousExecutionStateStep.execute(LoadPreviousExecutionStateStep.java:36)
        at org.gradle.internal.execution.steps.LoadPreviousExecutionStateStep.execute(LoadPreviousExecutionStateStep.java:23)
        at org.gradle.internal.execution.steps.HandleStaleOutputsStep.execute(HandleStaleOutputsStep.java:75)
        at org.gradle.internal.execution.steps.HandleStaleOutputsStep.execute(HandleStaleOutputsStep.java:41)
        at org.gradle.internal.execution.steps.AssignMutableWorkspaceStep.lambda$execute$0(AssignMutableWorkspaceStep.java:35)
        at org.gradle.api.internal.tasks.execution.TaskExecution$4.withWorkspace(TaskExecution.java:297)
        at org.gradle.internal.execution.steps.AssignMutableWorkspaceStep.execute(AssignMutableWorkspaceStep.java:31)
        at org.gradle.internal.execution.steps.AssignMutableWorkspaceStep.execute(AssignMutableWorkspaceStep.java:22)
        at org.gradle.internal.execution.steps.ChoosePipelineStep.execute(ChoosePipelineStep.java:40)
        at org.gradle.internal.execution.steps.ChoosePipelineStep.execute(ChoosePipelineStep.java:23)
        at org.gradle.internal.execution.steps.ExecuteWorkBuildOperationFiringStep.lambda$execute$2(ExecuteWorkBuildOperationFiringStep.java:67)
        at org.gradle.internal.execution.steps.ExecuteWorkBuildOperationFiringStep.execute(ExecuteWorkBuildOperationFiringStep.java:67)
        at org.gradle.internal.execution.steps.ExecuteWorkBuildOperationFiringStep.execute(ExecuteWorkBuildOperationFiringStep.java:39)
        at org.gradle.internal.execution.steps.IdentityCacheStep.execute(IdentityCacheStep.java:46)
        at org.gradle.internal.execution.steps.IdentityCacheStep.execute(IdentityCacheStep.java:34)
        at org.gradle.internal.execution.steps.IdentifyStep.execute(IdentifyStep.java:47)
        at org.gradle.internal.execution.steps.IdentifyStep.execute(IdentifyStep.java:34)
        at org.gradle.internal.execution.impl.DefaultExecutionEngine$1.execute(DefaultExecutionEngine.java:64)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeIfValid(ExecuteActionsTaskExecuter.java:132)
        ... 29 more
    Caused by: org.gradle.internal.service.ServiceCreationException: Could not create service of type IsolatedAntBuilder using BuildScopeServices.createIsolatedAntBuilder().       
        at org.gradle.internal.service.DefaultServiceRegistry$FactoryMethodService.invokeMethod(DefaultServiceRegistry.java:927)
        at org.gradle.internal.service.DefaultServiceRegistry$FactoryService.createServiceInstance(DefaultServiceRegistry.java:827)
        at org.gradle.internal.service.DefaultServiceRegistry$FactoryMethodService.createServiceInstance(DefaultServiceRegistry.java:945)
        at org.gradle.internal.service.DefaultServiceRegistry$ManagedObjectServiceProvider.getInstance(DefaultServiceRegistry.java:624)
        at org.gradle.internal.service.DefaultServiceRegistry$SingletonService.get(DefaultServiceRegistry.java:704)
        at org.gradle.internal.service.DefaultServiceRegistry.find(DefaultServiceRegistry.java:354)
        at org.gradle.internal.isolated.IsolationScheme$ServicesForIsolatedObject.find(IsolationScheme.java:238)
        at org.gradle.internal.isolated.IsolationScheme$ServicesForIsolatedObject.get(IsolationScheme.java:247)
        at org.gradle.internal.jacoco.JacocoReportAction$Inject.getIsolatedAntBuilder(Unknown Source)
        at org.gradle.api.plugins.internal.ant.AntWorkAction.execute(AntWorkAction.java:36)
        at org.gradle.workers.internal.DefaultWorkerServer.execute(DefaultWorkerServer.java:68)
        at org.gradle.workers.internal.AbstractClassLoaderWorker$1.create(AbstractClassLoaderWorker.java:54)
        at org.gradle.workers.internal.AbstractClassLoaderWorker$1.create(AbstractClassLoaderWorker.java:48)
        at org.gradle.internal.classloader.ClassLoaderUtils.executeInClassloader(ClassLoaderUtils.java:100)
        at org.gradle.workers.internal.AbstractClassLoaderWorker.executeInClassLoader(AbstractClassLoaderWorker.java:48)
        at org.gradle.workers.internal.IsolatedClassloaderWorker.run(IsolatedClassloaderWorker.java:49)
        at org.gradle.workers.internal.IsolatedClassloaderWorker.run(IsolatedClassloaderWorker.java:30)
        at org.gradle.workers.internal.IsolatedClassloaderWorkerFactory$1.lambda$execute$0(IsolatedClassloaderWorkerFactory.java:60)
        at org.gradle.workers.internal.AbstractWorker$1.call(AbstractWorker.java:44)
        at org.gradle.workers.internal.AbstractWorker$1.call(AbstractWorker.java:41)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:209)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:204)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:166)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:53)
        at org.gradle.workers.internal.AbstractWorker.executeWrappedInBuildOperation(AbstractWorker.java:41)
        at org.gradle.workers.internal.IsolatedClassloaderWorkerFactory$1.execute(IsolatedClassloaderWorkerFactory.java:52)
        at org.gradle.workers.internal.DefaultWorkerExecutor.lambda$submitWork$0(DefaultWorkerExecutor.java:176)
        at org.gradle.internal.work.DefaultConditionalExecutionQueue$ExecutionRunner.runExecution(DefaultConditionalExecutionQueue.java:194)
        at org.gradle.internal.work.DefaultConditionalExecutionQueue$ExecutionRunner.access$700(DefaultConditionalExecutionQueue.java:127)
        at org.gradle.internal.work.DefaultConditionalExecutionQueue$ExecutionRunner$1.run(DefaultConditionalExecutionQueue.java:169)
        at org.gradle.internal.Factories$1.create(Factories.java:31)
        at org.gradle.internal.work.DefaultWorkerLeaseService.withLocks(DefaultWorkerLeaseService.java:263)
        at org.gradle.internal.work.DefaultWorkerLeaseService.runAsWorkerThread(DefaultWorkerLeaseService.java:127)
        at org.gradle.internal.work.DefaultWorkerLeaseService.runAsWorkerThread(DefaultWorkerLeaseService.java:132)
        at org.gradle.internal.work.DefaultConditionalExecutionQueue$ExecutionRunner.runBatch(DefaultConditionalExecutionQueue.java:164)
        at org.gradle.internal.work.DefaultConditionalExecutionQueue$ExecutionRunner.run(DefaultConditionalExecutionQueue.java:133)
        ... 2 more
    Caused by: org.gradle.api.GradleException: Could not inspect the Groovy system for ClassLoader VisitableURLClassLoader(tooling-implementation-loader)
        at org.gradle.api.internal.classloading.GroovySystemLoaderFactory.forClassLoader(GroovySystemLoaderFactory.java:38)
        at org.gradle.api.internal.project.antbuilder.DefaultIsolatedAntBuilder.<init>(DefaultIsolatedAntBuilder.java:105)
        at org.gradle.internal.service.scopes.BuildScopeServices.createIsolatedAntBuilder(BuildScopeServices.java:390)
        at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:125)
        at org.gradle.internal.service.ReflectionBasedServiceMethod.invoke(ReflectionBasedServiceMethod.java:35)
        at org.gradle.internal.service.DefaultServiceRegistry$FactoryMethodService.invokeMethod(DefaultServiceRegistry.java:925)
        ... 40 more
    Caused by: java.lang.reflect.InaccessibleObjectException: Unable to make field private java.util.prefs.PreferenceChangeListener[] java.util.prefs.AbstractPreferences.prefListeners accessible: module java.prefs does not "opens java.util.prefs" to unnamed module @7a1ec503
        at org.gradle.api.internal.classloading.PreferenceCleaningGroovySystemLoader.<init>(PreferenceCleaningGroovySystemLoader.java:34)
        at org.gradle.api.internal.classloading.GroovySystemLoaderFactory.forClassLoader(GroovySystemLoaderFactory.java:35)
        ... 45 more


    BUILD FAILED in 15s
    14 actionable tasks: 14 executed
        at app//org.gradle.testkit.runner.internal.DefaultGradleRunner.lambda$build$2(DefaultGradleRunner.java:274)
        at app//org.gradle.testkit.runner.internal.DefaultGradleRunner.run(DefaultGradleRunner.java:367)
        at app//org.gradle.testkit.runner.internal.DefaultGradleRunner.build(DefaultGradleRunner.java:272)
        at app//io.github.aaravmahajanofficial.utils.TestProjectBuilder.runGradle(TestProjectBuilder.kt:268)
        at app//io.github.aaravmahajanofficial.utils.TestProjectBuilder.runGradle$default(TestProjectBuilder.kt:252)
        at app//io.github.aaravmahajanofficial.utils.TestProjectBuilder.runGradle(TestProjectBuilder.kt:250)
        at app//io.github.aaravmahajanofficial.QualityToolsIntegrationTest.execute jacoco with defaults(QualityToolsIntegrationTest.kt:364)

BUILD FAILED in 20s
7 actionable tasks: 2 executed, 5 up-to-date

### Expected Behavior

Build should be successful.

### Context (optional)

Looks this problem existed earlier, as well. Tried all the ways mentioned there, but no result.

https://github.com/gradle/gradle/issues/17229
https://github.com/gradle/gradle/issues/18508
https://github.com/jacoco/jacoco/issues/1234

---

Tried setting the jvmargs, org.gradle.daemon=true (or false) -> No result

"--add-opens=java.prefs/java.util.prefs=ALL-UNNAMED" +
  "--add-opens=java.base/java.lang.invoke=ALL-UNNAMED" +
  "--add-opens=java.base/java.lang=ALL-UNNAMED" +
  "--add-opens=java.base/java.io=ALL-UNNAMED" +
  "--add-opens=java.base/java.util=ALL-UNNAMED" +
  "--add-opens=java.base/java.lang.reflect=ALL-UNNAMED" +
  "--add-opens=java.base/java.security=ALL-UNNAMED" +
  "--add-opens=java.base/java.net=ALL-UNNAMED" +
  "--add-opens=java.base/sun.nio.ch=ALL-UNNAMED" +
  "--add-exports=java.base/sun.nio.ch=ALL-UNNAMED" +
  "--add-exports=jdk.unsupported/sun.misc=ALL-UNNAMED",

----

@oleg-nenashev , would appreciate your input on this issue

Thanks in advance!

### Self-contained Reproducer Project

https://github.com/aaravmahajanofficial/gradle-issue-34457-reproducer

### Gradle version

9.0-RC-4, 8.13, 8.14

### Build scan URL (optional)

_No response_

### Your Environment (optional)

------------------------------------------------------------
Gradle 9.0.0-rc-4
------------------------------------------------------------

Build time:    2025-07-28 15:10:02 UTC
Revision:      a0a6deb6f9937f9e9894a4460f4158267a13ad03

Kotlin:        2.2.0
Groovy:        4.0.27
Ant:           Apache Ant(TM) version 1.10.15 compiled on August 25 2024
Launcher JVM:  21.0.8 (Eclipse Adoptium 21.0.8+9-LTS)
Daemon JVM:    Compatible with Java 21, any vendor, nativeImageCapable=false (from gradle/gradle-daemon-jvm.properties)
OS:            Windows 11 10.0 amd64

> Caused by: java.lang.reflect.InaccessibleObjectException: Unable to make field private java.util.prefs.PreferenceChangeListener[] java.util.prefs.AbstractPreferences.prefListeners accessible: module java.prefs does not "opens java.util.prefs" to unnamed module @7a1ec503

So this comes from Java upgrade of course. The bytecode changed the behavior with the minimum version. I will need to see whether it is documented

--- a/platforms/core-runtime/base-services/src/main/java/org/gradle/internal/jvm/JpmsConfiguration.java
+++ b/platforms/core-runtime/base-services/src/main/java/org/gradle/internal/jvm/JpmsConfiguration.java
@@ -32,33 +32,14 @@ import java.util.List;
  * </p>
  */
 public class JpmsConfiguration {
-    private static final String ADD_OPENS_JAVA_PREFS_JAVA_UTIL_PREFS_ALL_UNNAMED = "--add-opens=java.prefs/java.util.prefs=ALL-UNNAMED";
-    private static final String ADD_OPENS_JAVA_BASE_JAVA_UTIL_ALL_UNNAMED = "--add-opens=java.base/java.util=ALL-UNNAMED";
-    private static final String ADD_OPENS_JAVA_BASE_JAVA_LANG_ALL_UNNAMED = "--add-opens=java.base/java.lang=ALL-UNNAMED";
-
-    /**
-     * Exposes the required packages for test workers to work on Java 9+.
-     */
-    private static final List<String> GRADLE_TEST_WORKERS_JPMS_ARGS_9 = ImmutableList.of(
-        ADD_OPENS_JAVA_BASE_JAVA_LANG_ALL_UNNAMED
-    );
-
-    /**
-     * Exposes the required packages for test workers to work on Java 9+.
-     */
-    private static final List<String> GRADLE_TEST_WORKERS_JPMS_IN_JAVA_GRADLE_PLUGIN_9 = ImmutableList.of(
-        ADD_OPENS_JAVA_BASE_JAVA_LANG_ALL_UNNAMED,
-        ADD_OPENS_JAVA_PREFS_JAVA_UTIL_PREFS_ALL_UNNAMED // required by PreferenceCleaningGroovySystemLoader
-    );
-
     /**
      * Exposes the required packages for Groovy to work on Java 9+.
      */
     private static final List<String> GROOVY_JPMS_ARGS_9 = ImmutableList.of(
-        ADD_OPENS_JAVA_BASE_JAVA_LANG_ALL_UNNAMED,
+        "--add-opens=java.base/java.lang=ALL-UNNAMED",
         "--add-opens=java.base/java.lang.invoke=ALL-UNNAMED",
-        ADD_OPENS_JAVA_BASE_JAVA_UTIL_ALL_UNNAMED,
-        ADD_OPENS_JAVA_PREFS_JAVA_UTIL_PREFS_ALL_UNNAMED, // required by PreferenceCleaningGroovySystemLoader
+        "--add-opens=java.base/java.util=ALL-UNNAMED",
+        "--add-opens=java.prefs/java.util.prefs=ALL-UNNAMED", // required by PreferenceCleaningGroovySystemLoader
         "--add-exports=jdk.compiler/com.sun.tools.javac.api=ALL-UNNAMED", // Required by JdkTools and JdkJavaCompiler
         "--add-exports=jdk.compiler/com.sun.tools.javac.util=ALL-UNNAMED" // Required by JdkTools and JdkJavaCompiler
     );
@@ -75,37 +56,30 @@ public class JpmsConfiguration {
      */
     private static final List<String> GRADLE_WORKER_JPMS_ARGS_24 = GRADLE_SHARED_JPMS_ARGS_24;
 
-    private static List<String> createDaemonJpmsArgs9() {
-        return ImmutableList.<String>builderWithExpectedSize(GROOVY_JPMS_ARGS_9.size() + 6)
-            .addAll(GROOVY_JPMS_ARGS_9)
-            .add("--add-opens=java.base/java.nio.charset=ALL-UNNAMED") // required by BeanSchemaKt
-            .add("--add-opens=java.base/java.net=ALL-UNNAMED") // required by JavaObjectSerializationCodec
-            .add("--add-opens=java.base/java.util.concurrent=ALL-UNNAMED") // required by AccessTrackingProperties
-            .add("--add-opens=java.base/java.util.concurrent.atomic=ALL-UNNAMED") // serialized from org.gradle.internal.file.StatStatistics$Collector
-            .add("--add-opens=java.xml/javax.xml.namespace=ALL-UNNAMED") // serialized from IvyDescriptorFileGenerator.Model
-            .add("--add-opens=java.base/java.time=ALL-UNNAMED").build(); // required by JavaObjectSerializationCodec
-    }
-
     /**
      * Exposes the required packages for the Gradle daemon to work on Java 9+.
      */
-    private static final List<String> GRADLE_DAEMON_JPMS_ARGS_9 = createDaemonJpmsArgs9();
+    private static final List<String> GRADLE_DAEMON_JPMS_ARGS_9 = ImmutableList.<String>builder()
+        .addAll(GROOVY_JPMS_ARGS_9)
+        .add("--add-opens=java.base/java.nio.charset=ALL-UNNAMED") // required by BeanSchemaKt
+        .add("--add-opens=java.base/java.net=ALL-UNNAMED") // required by JavaObjectSerializationCodec
+        .add("--add-opens=java.base/java.util.concurrent=ALL-UNNAMED") // required by AccessTrackingProperties
+        .add("--add-opens=java.base/java.util.concurrent.atomic=ALL-UNNAMED") // serialized from org.gradle.internal.file.StatStatistics$Collector
+        .add("--add-opens=java.xml/javax.xml.namespace=ALL-UNNAMED") // serialized from IvyDescriptorFileGenerator.Model
+        .add("--add-opens=java.base/java.time=ALL-UNNAMED")
+        .build();
 
 
     /**
      * Exposes the required packages for the Gradle daemon to work on Java 9+.
      * Also enables native access for the daemon process.
      */
-    private static final List<String> GRADLE_DAEMON_JPMS_ARGS_24 = createDaemonJpmsArgs24();
+    private static final List<String> GRADLE_DAEMON_JPMS_ARGS_24 = ImmutableList.<String>builder()
+        .addAll(GRADLE_DAEMON_JPMS_ARGS_9)
+        .addAll(GRADLE_SHARED_JPMS_ARGS_24)
+        .build();
 
-    private static List<String> createDaemonJpmsArgs24() {
-        return ImmutableList.<String>builderWithExpectedSize(GRADLE_DAEMON_JPMS_ARGS_9.size() + GRADLE_SHARED_JPMS_ARGS_24.size())
-            .addAll(GRADLE_DAEMON_JPMS_ARGS_9)
-            .addAll(GRADLE_SHARED_JPMS_ARGS_24)
-            .build();
-    }
-
-    public static List<String> forGroovyProcesses(int majorVersion) {
+    public static List<String> forGroovyCompilerWorker(int majorVersion) {
         if (majorVersion < 9) {
             return ImmutableList.of();
         }
@@ -141,31 +115,5 @@ public class JpmsConfiguration {
             return GRADLE_DAEMON_JPMS_ARGS_9;
         }
         return GRADLE_DAEMON_JPMS_ARGS_24;
-    }
-
-    /**
-     * Returns the JVM arguments that should be passed to a test worker process.
-     *
-     * @param majorVersion the major version of the JVM for the test worker process
-     * @return the JVM arguments
-     */
-    public static List<String> forTestWorkers(int majorVersion) {
-        if (majorVersion < 9) {
-            return ImmutableList.of();
-        }
-        return GRADLE_TEST_WORKERS_JPMS_ARGS_9;
-    }
-
-    /**
-     * Returns the JVM arguments that should be passed to a test worker process when using TestKit.
-     *
-     * @param majorVersion the major version of the JVM for the test worker process
-     * @return the JVM arguments
-     */
-    public static List<String> forTestWorkersInJavaGradlePlugin(int majorVersion) {
-        if (majorVersion < 9) {
-            return ImmutableList.of();
-        }
-        return GRADLE_TEST_WORKERS_JPMS_IN_JAVA_GRADLE_PLUGIN_9;
     }
 }

--- a/platforms/core-runtime/base-services/src/test/groovy/org/gradle/internal/jvm/JpmsConfigurationTest.groovy
+++ b/platforms/core-runtime/base-services/src/test/groovy/org/gradle/internal/jvm/JpmsConfigurationTest.groovy
@@ -22,7 +22,7 @@ class JpmsConfigurationTest extends Specification {
 
     def "forGroovyProcesses returns empty list for Java 8 and below"() {
         expect:
-        JpmsConfiguration.forGroovyProcesses(majorVersion) == []
+        JpmsConfiguration.forGroovyCompilerWorker(majorVersion) == []
 
         where:
         majorVersion << [1, 5, 6, 7, 8]
@@ -30,16 +30,17 @@ class JpmsConfigurationTest extends Specification {
 
     def "forGroovyProcesses returns JPMS args for Java 9 and above"() {
         when:
-        def result = JpmsConfiguration.forGroovyProcesses(majorVersion)
+        def result = JpmsConfiguration.forGroovyCompilerWorker(majorVersion)
 
         then:
-        result.size() == 6
-        result.contains("--add-opens=java.base/java.lang=ALL-UNNAMED")
-        result.contains("--add-opens=java.base/java.lang.invoke=ALL-UNNAMED")
-        result.contains("--add-opens=java.base/java.util=ALL-UNNAMED")
-        result.contains("--add-opens=java.prefs/java.util.prefs=ALL-UNNAMED")
-        result.contains("--add-exports=jdk.compiler/com.sun.tools.javac.api=ALL-UNNAMED")
-        result.contains("--add-exports=jdk.compiler/com.sun.tools.javac.util=ALL-UNNAMED")
+        result == [
+            "--add-opens=java.base/java.lang=ALL-UNNAMED",
+            "--add-opens=java.base/java.lang.invoke=ALL-UNNAMED",
+            "--add-opens=java.base/java.util=ALL-UNNAMED",
+            "--add-opens=java.prefs/java.util.prefs=ALL-UNNAMED",
+            "--add-exports=jdk.compiler/com.sun.tools.javac.api=ALL-UNNAMED",
+            "--add-exports=jdk.compiler/com.sun.tools.javac.util=ALL-UNNAMED",
+        ]
 
         where:
         majorVersion << [9, 11, 17, 21, 24, 25]
@@ -85,21 +86,20 @@ class JpmsConfigurationTest extends Specification {
         def result = JpmsConfiguration.forDaemonProcesses(majorVersion, false)
 
         then:
-        result.size() == 12
-        // Groovy JPMS args
-        result.contains("--add-opens=java.base/java.lang=ALL-UNNAMED")
-        result.contains("--add-opens=java.base/java.lang.invoke=ALL-UNNAMED")
-        result.contains("--add-opens=java.base/java.util=ALL-UNNAMED")
-        result.contains("--add-opens=java.prefs/java.util.prefs=ALL-UNNAMED")
-        result.contains("--add-exports=jdk.compiler/com.sun.tools.javac.api=ALL-UNNAMED")
-        result.contains("--add-exports=jdk.compiler/com.sun.tools.javac.util=ALL-UNNAMED")
-        // Additional daemon JPMS args
-        result.contains("--add-opens=java.base/java.nio.charset=ALL-UNNAMED")
-        result.contains("--add-opens=java.base/java.net=ALL-UNNAMED")
-        result.contains("--add-opens=java.base/java.util.concurrent=ALL-UNNAMED")
-        result.contains("--add-opens=java.base/java.util.concurrent.atomic=ALL-UNNAMED")
-        result.contains("--add-opens=java.xml/javax.xml.namespace=ALL-UNNAMED")
-        result.contains("--add-opens=java.base/java.time=ALL-UNNAMED")
+        result == [
+            "--add-opens=java.base/java.lang=ALL-UNNAMED",
+            "--add-opens=java.base/java.lang.invoke=ALL-UNNAMED",
+            "--add-opens=java.base/java.util=ALL-UNNAMED",
+            "--add-opens=java.prefs/java.util.prefs=ALL-UNNAMED",
+            "--add-exports=jdk.compiler/com.sun.tools.javac.api=ALL-UNNAMED",
+            "--add-exports=jdk.compiler/com.sun.tools.javac.util=ALL-UNNAMED",
+            "--add-opens=java.base/java.nio.charset=ALL-UNNAMED",
+            "--add-opens=java.base/java.net=ALL-UNNAMED",
+            "--add-opens=java.base/java.util.concurrent=ALL-UNNAMED",
+            "--add-opens=java.base/java.util.concurrent.atomic=ALL-UNNAMED",
+            "--add-opens=java.xml/javax.xml.namespace=ALL-UNNAMED",
+            "--add-opens=java.base/java.time=ALL-UNNAMED",
+        ]
 
         where:
         majorVersion << [9, 11, 17, 21, 23]
@@ -111,9 +111,20 @@ class JpmsConfigurationTest extends Specification {
 
         then:
         // Should be same as without native services for Java < 24
-        result.size() == 12
-        result.contains("--add-opens=java.base/java.util=ALL-UNNAMED")
-        !result.contains("--enable-native-access=ALL-UNNAMED")
+        result == [
+            "--add-opens=java.base/java.lang=ALL-UNNAMED",
+            "--add-opens=java.base/java.lang.invoke=ALL-UNNAMED",
+            "--add-opens=java.base/java.util=ALL-UNNAMED",
+            "--add-opens=java.prefs/java.util.prefs=ALL-UNNAMED",
+            "--add-exports=jdk.compiler/com.sun.tools.javac.api=ALL-UNNAMED",
+            "--add-exports=jdk.compiler/com.sun.tools.javac.util=ALL-UNNAMED",
+            "--add-opens=java.base/java.nio.charset=ALL-UNNAMED",
+            "--add-opens=java.base/java.net=ALL-UNNAMED",
+            "--add-opens=java.base/java.util.concurrent=ALL-UNNAMED",
+            "--add-opens=java.base/java.util.concurrent.atomic=ALL-UNNAMED",
+            "--add-opens=java.xml/javax.xml.namespace=ALL-UNNAMED",
+            "--add-opens=java.base/java.time=ALL-UNNAMED",
+        ]
 
         where:
         majorVersion << [9, 11, 17, 21, 23]
@@ -124,20 +135,21 @@ class JpmsConfigurationTest extends Specification {
         def result = JpmsConfiguration.forDaemonProcesses(majorVersion, true)
 
         then:
-        result.size() == 13
-        // Groovy JPMS args
-        result.contains("--add-opens=java.base/java.lang=ALL-UNNAMED")
-        result.contains("--add-opens=java.base/java.util=ALL-UNNAMED")
-        result.contains("--add-opens=java.prefs/java.util.prefs=ALL-UNNAMED")
-        // Additional daemon JPMS args
-        result.contains("--add-opens=java.base/java.nio.charset=ALL-UNNAMED")
-        result.contains("--add-opens=java.base/java.net=ALL-UNNAMED")
-        result.contains("--add-opens=java.base/java.util.concurrent=ALL-UNNAMED")
-        result.contains("--add-opens=java.base/java.util.concurrent.atomic=ALL-UNNAMED")
-        result.contains("--add-opens=java.xml/javax.xml.namespace=ALL-UNNAMED")
-        result.contains("--add-opens=java.base/java.time=ALL-UNNAMED")
-        // Native access flag
-        result.contains("--enable-native-access=ALL-UNNAMED")
+        result == [
+            "--add-opens=java.base/java.lang=ALL-UNNAMED",
+            "--add-opens=java.base/java.lang.invoke=ALL-UNNAMED",
+            "--add-opens=java.base/java.util=ALL-UNNAMED",
+            "--add-opens=java.prefs/java.util.prefs=ALL-UNNAMED",
+            "--add-exports=jdk.compiler/com.sun.tools.javac.api=ALL-UNNAMED",
+            "--add-exports=jdk.compiler/com.sun.tools.javac.util=ALL-UNNAMED",
+            "--add-opens=java.base/java.nio.charset=ALL-UNNAMED",
+            "--add-opens=java.base/java.net=ALL-UNNAMED",
+            "--add-opens=java.base/java.util.concurrent=ALL-UNNAMED",
+            "--add-opens=java.base/java.util.concurrent.atomic=ALL-UNNAMED",
+            "--add-opens=java.xml/javax.xml.namespace=ALL-UNNAMED",
+            "--add-opens=java.base/java.time=ALL-UNNAMED",
+            "--enable-native-access=ALL-UNNAMED"
+        ]
 
         where:
         majorVersion << [24, 25, 26]
@@ -148,113 +160,22 @@ class JpmsConfigurationTest extends Specification {
         def result = JpmsConfiguration.forDaemonProcesses(majorVersion, false)
 
         then:
-        result.size() == 12
-        !result.contains("--enable-native-access=ALL-UNNAMED")
+        result == [
+            "--add-opens=java.base/java.lang=ALL-UNNAMED",
+            "--add-opens=java.base/java.lang.invoke=ALL-UNNAMED",
+            "--add-opens=java.base/java.util=ALL-UNNAMED",
+            "--add-opens=java.prefs/java.util.prefs=ALL-UNNAMED",
+            "--add-exports=jdk.compiler/com.sun.tools.javac.api=ALL-UNNAMED",
+            "--add-exports=jdk.compiler/com.sun.tools.javac.util=ALL-UNNAMED",
+            "--add-opens=java.base/java.nio.charset=ALL-UNNAMED",
+            "--add-opens=java.base/java.net=ALL-UNNAMED",
+            "--add-opens=java.base/java.util.concurrent=ALL-UNNAMED",
+            "--add-opens=java.base/java.util.concurrent.atomic=ALL-UNNAMED",
+            "--add-opens=java.xml/javax.xml.namespace=ALL-UNNAMED",
+            "--add-opens=java.base/java.time=ALL-UNNAMED",
+        ]
 
         where:
         majorVersion << [24, 25, 26]
-    }
-
-    def "forTestWorkers returns empty list for Java 8 and below"() {
-        expect:
-        JpmsConfiguration.forTestWorkers(majorVersion) == []
-
-        where:
-        majorVersion << [1, 5, 6, 7, 8]
-    }
-
-    def "forTestWorkers returns JPMS args for Java 9 and above"() {
-        when:
-        def result = JpmsConfiguration.forTestWorkers(majorVersion)
-
-        then:
-        result.size() == 1
-        result.contains("--add-opens=java.base/java.lang=ALL-UNNAMED")
-
-        where:
-        majorVersion << [9, 11, 17, 21, 24, 25]
-    }
-
-    def "forTestWorkersWithTestKit returns JPMS args for Java 9 and above"() {
-        when:
-        def result = JpmsConfiguration.forTestWorkersInJavaGradlePlugin(majorVersion)
-
-        then:
-        result.size() == 2
-        result.contains("--add-opens=java.base/java.lang=ALL-UNNAMED")
-        result.contains("--add-opens=java.prefs/java.util.prefs=ALL-UNNAMED")
-
-        where:
-        majorVersion << [9, 11, 17, 21, 24, 25]
-    }
-
-    def "forTestWorkersWithTestKit returns empty list for Java 8 and below"() {
-        expect:
-        JpmsConfiguration.forTestWorkersInJavaGradlePlugin(majorVersion) == []
-
-        where:
-        majorVersion << [1, 5, 6, 7, 8]
-    }
-
-    def "all methods return immutable lists"() {
-        when:
-        def groovyResult = JpmsConfiguration.forGroovyProcesses(9)
-        def workerResult = JpmsConfiguration.forWorkerProcesses(24, true)
-        def daemonResult = JpmsConfiguration.forDaemonProcesses(9, false)
-        def testWorkerResult = JpmsConfiguration.forTestWorkers(9)
-        def testWorkerWithTestKitResult = JpmsConfiguration.forTestWorkersInJavaGradlePlugin(9)
-
-        then:
-        // Attempting to modify should throw UnsupportedOperationException
-        [groovyResult, workerResult, daemonResult, testWorkerResult, testWorkerWithTestKitResult].each { list ->
-            try {
-                list.add("--some-arg")
-                assert false, "Expected UnsupportedOperationException"
-            } catch (UnsupportedOperationException e) {
-                // Expected
-            }
-        }
-    }
-
-    def "forTestWorkers returns consistent results across versions"() {
-        expect:
-        JpmsConfiguration.forTestWorkers(9) == JpmsConfiguration.forTestWorkers(17)
-        JpmsConfiguration.forTestWorkers(17) == JpmsConfiguration.forTestWorkers(24)
-    }
-
-    def "forTestWorkersWithTestKit returns consistent results across versions"() {
-        expect:
-        JpmsConfiguration.forTestWorkersInJavaGradlePlugin(9) == JpmsConfiguration.forTestWorkersInJavaGradlePlugin(17)
-        JpmsConfiguration.forTestWorkersInJavaGradlePlugin(17) == JpmsConfiguration.forTestWorkersInJavaGradlePlugin(24)
-    }
-
-    def "all JPMS args lists are non-null"() {
-        expect:
-        JpmsConfiguration.forGroovyProcesses(version) != null
-        JpmsConfiguration.forWorkerProcesses(version, nativeServices) != null
-        JpmsConfiguration.forDaemonProcesses(version, nativeServices) != null
-        JpmsConfiguration.forTestWorkers(version) != null
-        JpmsConfiguration.forTestWorkersInJavaGradlePlugin(version) != null
-
-        where:
-        version | nativeServices
-        8       | false
-        8       | true
-        9       | false
-        9       | true
-        17      | false
-        17      | true
-        24      | false
-        24      | true
-    }
-
-    def "forTestWorkers contains subset of forTestWorkersWithTestKit args"() {
-        when:
-        def testWorkersArgs = JpmsConfiguration.forTestWorkers(9)
-        def testWorkersWithTestKitArgs = JpmsConfiguration.forTestWorkersInJavaGradlePlugin(9)
-
-        then:
-        testWorkersWithTestKitArgs.containsAll(testWorkersArgs)
-        testWorkersWithTestKitArgs.size() > testWorkersArgs.size()
     }
 }

--- a/platforms/extensibility/plugin-development/src/main/java/org/gradle/plugin/devel/plugins/JavaGradlePluginPlugin.java
+++ b/platforms/extensibility/plugin-development/src/main/java/org/gradle/plugin/devel/plugins/JavaGradlePluginPlugin.java
@@ -450,7 +450,7 @@ public abstract class JavaGradlePluginPlugin implements Plugin<Project> {
 
             project.getTasks().withType(Test.class).configureEach(test -> {
                 test.getInputs()
-                    .files(pluginClasspathTask.get().getPluginClasspath())
+                    .files(pluginClasspathTask.map(PluginUnderTestMetadata::getPluginClasspath))
                     .withPropertyName("pluginClasspath")
                     .withNormalizer(ClasspathNormalizer.class);
 
@@ -481,7 +481,7 @@ public abstract class JavaGradlePluginPlugin implements Plugin<Project> {
         @Override
         public Iterable<String> asArguments() {
             int majorVersion = Integer.parseInt(test.getJavaVersion().getMajorVersion());
-            return JpmsConfiguration.forTestWorkersInJavaGradlePlugin(majorVersion);
+            return JpmsConfiguration.forDaemonProcesses(majorVersion, true);
         }
     }
 

--- a/platforms/jvm/language-groovy/src/main/java/org/gradle/api/internal/tasks/compile/daemon/DaemonGroovyCompiler.java
+++ b/platforms/jvm/language-groovy/src/main/java/org/gradle/api/internal/tasks/compile/daemon/DaemonGroovyCompiler.java
@@ -112,7 +112,7 @@ public class DaemonGroovyCompiler extends AbstractDaemonCompiler<GroovyJavaJoint
         javaForkOptions.setWorkingDir(daemonWorkingDir);
         javaForkOptions.setExecutable(javaOptions.getExecutable());
         int javaVersionMajor = jvmVersionDetector.getJavaVersionMajor(javaOptions.getExecutable());
-        javaForkOptions.jvmArgs(JpmsConfiguration.forGroovyProcesses(javaVersionMajor));
+        javaForkOptions.jvmArgs(JpmsConfiguration.forGroovyCompilerWorker(javaVersionMajor));
         if (javaVersionMajor <= 8) {
             // In JDK 8 and below, we need to attach the 'tools.jar' to the classpath.
             File javaExecutable = new File(javaForkOptions.getExecutable());

